pkg/operators/matchers/regexcache.go

```diff
@@ -0,0 +1,133 @@
+package matchers
+
+import (
+	"regexp"
+	"sync"
+	"time"
+)
+
+type regexCacheKey struct {
+	pattern string
+	corpus  string
+}
+
+type regexCacheEntry struct {
```

Comment:
Do we really need to cache matches and does it actually improve because we are storing matched and not sure if its required (as we also incur storage overhead) 

We can do some benchmarks / synthetic tests to identigy whether it improves on standard nuclei-templates run